packages/ember-glimmer/lib/utils/references.js

```diff
@@ -220,7 +220,7 @@ export class NestedPropertyReference extends PropertyReference {
       return parentValue.length;
     }
 
-    if (typeof parentValue === 'object' && parentValue) {
+    if ((typeof parentValue === 'object' || typeof parentValue === 'function') && parentValue) {
```

Comment:
Falsy check should have been !== null and it needs to stay together with the object check. Same with the following. Don't put the function check in between it's confusing it's either an object that isn't null or a function